libs/trainer/bbhnet/trainer/trainer.py

```diff
@@ -271,4 +283,23 @@ def train(
                     )
                     break
 
+        # if no validation dataset passed, save model based on
+        # best training loss. i.e. attempt to overfit
+        else:
+            # update our learning rate scheduler if we
+            # indicated a schedule with `patience`
+            if patience is not None:
+                lr_scheduler.step(train_loss)
+
+            if train_loss < best_train_loss:
+                logging.debug(
+                    "No validation dataset passed. "
+                    "Achieved new lowest training loss, "
+                    "saving model weights "
+                )
+                best_train_loss = train_loss
+                weights_path = os.path.join(outdir, "weights.pt")
+                torch.save(model.state_dict(), weights_path)
+
+    print(history)
```

Comment:
Is this a leftover debugging artifact or do you want this for the purposes of reaching pinto's stdout?